src/libraries/System.Collections/tests/BitArray/BitArray_GetSetTests.cs

```diff
@@ -552,5 +552,37 @@ public static void CopyTo_Byte_Hidden(string label, BitArray bits)
                 Assert.Equal((byte)((1 << remainder) - 1), data[fullBytes]);
             }
         }
+
+        [Theory]
+        [MemberData(nameof(HasAllSet_TestData))]
+        public static void HasAllSet(bool[] bits, bool expectedResult)
+        {
+            var bitArray = new BitArray(bits);
+            Assert.Equal(expectedResult, bitArray.HasAllSet());
+        }
+
+        [Theory]
+        [MemberData(nameof(HasAnySet_TestData))]
+        public static void HasAnySet(bool[] bits, bool expectedResult)
+        {
+            var bitArray = new BitArray(bits);
+            Assert.Equal(expectedResult, bitArray.HasAnySet());
+        }
+
+        public static IEnumerable<object[]> HasAllSet_TestData() => HasAnySet_TestData().Select(parameters => new object[] { ((bool[])parameters[0]).Select(bit => !bit).ToArray(), !(bool)parameters[1] });
+
+        public static IEnumerable<object[]> HasAnySet_TestData()
+        {
+            yield return new object[] { Array.Empty<bool>(), false };
+
+            foreach (int size in new[] { 1, BitsPerInt32 - 1, BitsPerInt32, BitsPerInt32 + 1, BitsPerInt32 * 2 - 1 })
+            {
+                yield return new object[] { new bool[size], false };
+                
+                bool[] bits = new bool[size];
+                bits[^1] = true;
```

Comment:
These tests are only testing where the last bit is true (or in the all case, where the last bit is false).  There are thus no tests validating HasAllSet returning true (other than the empty case), and there are no tests validating when other bits in the pattern are on/off besides the last one.

Reply:
> There are thus no tests validating HasAllSet returning true (other than the empty case)

Actually "validating HasAllSet returning true" cases were already covered just by Line 580 `yield return new object[] { new bool[size], false };` with help of `HasAllSet_TestData()` method


> there are no tests validating when other bits in the pattern are on/off besides the last one.

Added cases of all other single on/off bits besides of the last bit. (I didn't add all multiple bits on/off combination considering larger test sample data size.)